cluster-controller/src/main/java/io/strimzi/controller/cluster/resources/Source2Image.java

```diff
@@ -0,0 +1,212 @@
+package io.strimzi.controller.cluster.resources;
+
+import io.fabric8.kubernetes.api.model.ObjectReference;
+import io.fabric8.openshift.api.model.BinaryBuildSource;
+import io.fabric8.openshift.api.model.BuildConfig;
+import io.fabric8.openshift.api.model.BuildConfigBuilder;
+import io.fabric8.openshift.api.model.BuildTriggerPolicy;
+import io.fabric8.openshift.api.model.ImageChangeTrigger;
+import io.fabric8.openshift.api.model.ImageLookupPolicyBuilder;
+import io.fabric8.openshift.api.model.ImageStream;
+import io.fabric8.openshift.api.model.ImageStreamBuilder;
+import io.fabric8.openshift.api.model.TagReference;
+import io.vertx.core.json.JsonObject;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Represents S2I upgrade on top of the regular resource
+ */
+public class Source2Image {
+    private final String name;
+    private final String namespace;
+    private final String sourceImage;
+    private final String tag;
+    private final String targetImage;
+    protected Map<String, String> labels;
+
+    // Annotations
+    public static String ANNOTATION_S2I = "s2i";
+    public static String ANNOTATION_RESOLVE_NAMES = "alpha.image.policy.openshift.io/resolve-names";
+
+    // Keys to config JSON
+    private static String KEY_SOURCE_IMAGE = "sourceImage";
+    public static String KEY_ENABLED = "enabled";
+
+    // Default values
+    private static String DEFAULT_SOURCE_IMAGE = "strimzi/kafka-connect-s2i:latest";
+
+    /**
+     * Constructor
+     *
+     * @param namespace     OpenShift project
+     * @param name       Name od the name
```

Comment:
typo "od" but "Name of the name" ? What does it mean ? :-) Is "name" the cluster ?